lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -146,7 +155,7 @@ public void writeBlock(ColumnChunk columnChunk)
         }
 
         long currentPageBufferedBytes = getCurrentPageBufferedBytes();
-        if (currentPageBufferedBytes >= pageSizeThreshold) {
+        if (valueCount >= pageValueCountLimit || currentPageBufferedBytes >= pageSizeThreshold) {
```

Comment:
Probably the gist of this PR is found on this check from a technical standpoint.
It is not clear for me (also after skimming the Slack thread mentioned in the description) though why do we need this extra check  next to the `pageSizeThreshold` for performing the flushing.

Can you pls add to the description of the PR what is the logical reasoning pushing for these changes?

Reply:
I think tuning `pageSizeThreshold` might also have worked in that case. However, the current default of 1MB is already what other parquet writers based on parquet-mr have also been using. Both ORC and parquet-mr have page value count limits of 10K and 20K respectively though and we're not applying any value count limit in our parquet writer at the moment. We're going to need this config in future anyway when we implement writing page level indexes as the indexes become less effective with large value count pages.

Reply:
Given that max is 1MB (which seems pretty small), why there is a need for another limit?